microservice/src/test/java/de/gsi/microservice/concepts/majordomo/legacy/CmwLightClient.java

```diff
@@ -0,0 +1,319 @@
+    public static final byte RT_EXCEPTION = 4;
+    public static final byte RT_SUBSCRIBE = 5;
+    public static final byte RT_UNSUBSCRIBE = 6;
+    public static final byte RT_NOTIFICATION_DATA = 7;
+    public static final byte RT_NOTIFICATION_EXC = 8;
+    public static final byte RT_SUBSCRIBE_EXCEPTION = 9;
+    public static final byte RT_EVENT = 10; // Also used as close
+    public static final byte RT_SESSION_CONFIRM = 11;
+
+    // UpdateType
+    public static final byte UT_NORMAL = (byte) 0;
+    public static final byte UT_FIRST_UPDATE = (byte) 1; // Initial update sent when the subscription is created.
+    public static final byte UT_IMMEDIATE_UPDATE = (byte) 2; //Update sent after the value has been modified by a set call.
+
+    public CmwLightClient() {
+        System.out.println("create new context");
+        controlChannel = context.createSocket(SocketType.DEALER);
+        System.out.println("setup socket");
+        controlChannel.setSndHWM(0);
+        controlChannel.setRcvHWM(0);
+        controlChannel.setIdentity(getIdentity().getBytes()); // hostname/process/id/channel
+        controlChannel.setLinger(0);
+    }
+
+    private String getIdentity() {
+        String hostname;
+        try {
+            hostname = InetAddress.getLocalHost().getHostName();
+        } catch (UnknownHostException e) {
+            hostname = "localhost";
+        }
+        final long processId = ProcessHandle.current().pid();
+        final long connectionId = 1L;
+        final long channelId = 1L;
+        return hostname + '/' + processId + '/' + connectionId + '/' + channelId;
+    }
+
+    private void receiveData() {
+        System.out.println();
+        final ZMsg data = ZMsg.recvMsg(controlChannel);
+        final ZFrame firstFrame = data.pollFirst();
+        if (Arrays.equals(firstFrame.getData(), new byte[] { SERVER_HB })) {
+            System.out.println("Heartbeat received");
+            // TODO: reset Heartbeat
+            return;
+        }
+        if (!Arrays.equals(firstFrame.getData(), new byte[] { SERVER_REP })) {
+            System.out.println("expected reply message, but got: " + data);
+        }
+        final byte[] descriptor = data.pollLast().getData();
+        if (Arrays.equals(data.getLast().getData(), new byte[] { MT_HEADER })) {
+            System.out.println("received header: " + data);
+        } else if (Arrays.equals(data.getLast().getData(), new byte[] { MT_HEADER, MT_BODY_EXCEPTION })) {
+            System.out.println("received exception: " + data);
+        } else if (!Arrays.equals(data.getLast().getData(), new byte[] { MT_HEADER, MT_BODY, MT_BODY_DATA_CONTEXT })) {
+            System.out.println("expected reply message, but got: " + Arrays.toString(data.getLast().getData()));
+            return;
+        } else {
+            System.out.println("received reply: " + data);
+        }
+        for (final byte desc : descriptor) {
+            switch (desc) {
```

Comment:
Codacy found an issue: [A switch statement does not contain a break](https://app.codacy.com/manual/GSI/chart-fx/pullRequest?prid=6307742)